articles/azure-monitor/app/tutorial-asp-net-core.md

```diff
@@ -0,0 +1,525 @@
+
+1. From the Visual Studio Solution Explorer, locate and open the **Program.cs** file.
+
+    ![The Visual Studio Solution Explorer displays with the Program.cs highlighted.](./media/asp-net-core/solution_explorer_programcs.png "Program.cs")
+
+2. Insert the following code prior to the `builder.Services.AddControllersWithViews()` statement. This code automatically reads the Application Insights connection string value from configuration (this can be from the `appsettings.json` file or Azure App Service Configuration). The `AddApplicationInsightsTelemetry` method registers the `ApplicationInsightsLoggerProvider` with the built-in dependency injection container, that will then be used to fulfill [ILogger](https://docs.microsoft.com/dotnet/api/microsoft.extensions.logging.ilogger) and [ILogger\<TCategoryName\>](https://docs.microsoft.com/dotnet/api/microsoft.extensions.logging.iloggerprovider) implementation requests.
+
+    ```csharp
+    builder.Services.AddApplicationInsightsTelemetry();
+    ```
+
+    ![A code window displays with the preceding code snippet highlighted.](./media/asp-net-core/enable_server_side_telemetry.png "Enable server-side telemetry")
+
+    > [!TIP]
+    > Learn more about [configuration options in ASP.NET Core](https://docs.microsoft.com/aspnet/core/fundamentals/configuration/?view=aspnetcore-6.0).
+
+## Enable client-side telemetry for web applications
+
+The preceding steps are enough to help you start collecting server-side telemetry. If your application has client-side components, follow the next steps to start collecting [usage telemetry](./usage-overview.md).
+
+1. In Visual Studio Solution explorer, locate and open `\Views\_ViewImports.cshtml`. Add the following code at the end of the existing file.
+
+    ```cshtml
+    @inject Microsoft.ApplicationInsights.AspNetCore.JavaScriptSnippet JavaScriptSnippet
+    ```
+
+    ![The _ViewImports.cshtml file displays with the preceding line of code highlighted.](./media/asp-net-core/view_imports_injection.png "JavaScriptSnippet injection")
+
+2. To properly enable client-side monitoring for your application, the JavaScript snippet must appear in the `<head>` section of each page of your application that you want to monitor. In Visual Studio Solution Explorer, locate and open  `\Views\Shared\_Layout.cshtml`, insert the following code immediately preceding the closing `<\head>` tag.
+
+    ```cshtml
+    @Html.Raw(JavaScriptSnippet.FullScript)
+    ```
+
+    ![The _Layout.cshtml file displays with the preceding line of code highlighted within the head section of the page.](./media/asp-net-core/layout_head_code.png "The head section of _Layout.cshtml")
+
+    > [!TIP]
+    > As an alternative to using the `FullScript`, the `ScriptBody` is available starting in Application Insights SDK for ASP.NET Core version 2.14. Use `ScriptBody` if you need to control the `<script>` tag to set a Content Security Policy:
```

Comment:
I'd suggest to always assume user is on latest SDK, so no need of mentioning "starting with so and so version". This will make smoother reading experience

Reply:
Fixed